src/cfi.rs

```diff
@@ -4885,4 +4941,31 @@ mod tests {
         let mut iter = section.entries(&bases);
         assert_eq!(iter.next(), Err(Error::FuncRelativePointerInBadContext));
     }
+
+    #[test]
+    fn register_rule_map_eq() {
+        // Different order, but still equal.
+        let map1: RegisterRuleMap<LittleEndian> = [(0, RegisterRule::SameValue),
+                                                   (3, RegisterRule::Offset(1))].iter().collect();
+        let map2: RegisterRuleMap<LittleEndian> = [(3, RegisterRule::Offset(1)),
+                                                   (0, RegisterRule::SameValue)].iter().collect();
+        assert_eq!(map1, map2);
+        assert_eq!(map2, map1);
+
+        // Not equal.
+        let map3: RegisterRuleMap<LittleEndian> = [(0, RegisterRule::SameValue),
+                                                   (2, RegisterRule::Offset(1))].iter().collect();
+        let map4: RegisterRuleMap<LittleEndian> = [(3, RegisterRule::Offset(1)),
+                                                   (0, RegisterRule::SameValue)].iter().collect();
+        assert!(map3 != map4);
+        assert!(map4 != map3);
+
+        // One has undefined, other implicitly has undefined.
+        let mut map5 = RegisterRuleMap::<LittleEndian>::default();
+        map5.set(0, RegisterRule::SameValue);
+        map5.set(0, RegisterRule::Undefined);
```

Comment:
Not sure I like that `set` treats undefined differently based on whether there was already a value. I can see why you've done it though. If we're going to do that, I think we should hide the difference from the user, so we should at least make `get` return `None` if the rule is undefined, whether implicit or not.